Create rule S6862: Beans in "@Configuration" class should have different names (#3474)
SonarSource/rspec: rules/S6862/java/rule.adoc

== Why is this an issue?

Naming conventions play a crucial role in maintaining code clarity and readability.

> In Spring configurations, the uniqueness of bean names is vital to avoid confusion for both developers and the Spring Context.

Reviewer: We could change this sentence a bit to fit better together with my suggestion from below, and also to make it shorter. Also, "To avoid confusion for ... the Spring Context" sounds somewhat strange (you cannot confuse the context). I would just skip it.

```suggestion
The uniqueness of bean names in Spring configurations is vital to the clarity and readability of the code.
```

> When two beans share the same name within a configuration, it becomes challenging for a reader to discern which bean is being referred to, leading to potential misunderstandings and errors.

Reviewer: I would prefer a simpler language here as not to put a barrier for non-native speakers when they read our RSPECs. I think "discern" is sophisticated vocabulary and "decide" or "distinguish" might be simpler. We could also restructure the sentence a bit:

```suggestion
When two beans share the same name within a configuration, it is not obvious to the reader which bean is being referred to. This leads to potential misunderstandings and errors.
```

== How to fix it

To address this issue, ensure each bean within a configuration has a distinct and meaningful name.

> Choose names that accurately represent the purpose or functionality of the bean, making it easier for developers to understand the role of each component.

Reviewer: Maybe the second part: "making it easier for developers to understand the role of each component" is a bit redundant and could be skipped, but I have no strong opinion here.
